### ISSUE TITLE
corners_to_keypoints fails when no corners detected

### ISSUE THREAD

**Issue by AdamSpannbauer** [jrosebr1/imutils#71]
## Overview

The `corners_to_keypoints` utility does not perform any checks on the input to ensure it is an iterable as expected.  This can cause error if using a GFTT keypoint detector on an image that doesn't have any keypoints found (ie `corners_to_keypoints(None)`).

(I ran into the issue with all black images from a VideoCapture while the camera was starting up)

## Example to reproduce error

```python
import cv2
import numpy as np
import imutils.feature.factories as kp_factory

# create blank all black 'image'
image = np.zeros(16, dtype='uint8').reshape(4, 4)

# init gftt keypoint detector using imutils factory
kp_detector = kp_factory.FeatureDetector_create('GFTT')

# detect keypoints in blank image (same defaults as `FeatureDetector_create('GFTT')`)
# result assigned to cv2_kps is `None`
cv2_kps = cv2.goodFeaturesToTrack(image, 0, 0.01, 1)  

# attempt keypoint detection with imutils (causes error below)
imutils_kps = kp_detector.detect(image)
```

### Resulting error

```python
Traceback (most recent call last):
   File "test_kp_none.py", line 12, in <module>
      imutils_kps = kp_detector.detect(image)
   File ".../imutils/imutils/feature/gftt.py", line 21, in detect
      return corners_to_keypoints(cnrs)
   File ".../imutils/imutils/feature/helpers.py", line 6, in corners_to_keypoints
      return [cv2.KeyPoint(kp[0][0], kp[0][1], 1) for kp in corners]
TypeError: 'NoneType' object is not iterable
```

## Proposed fix

A simple if else to return ~`None`~ `[]` if the input to `corners_to_keypoints` is `None`.  This fixes the error and ~seems to mirror the `cv2` behavior~ is consistent with the other imutils kp detectors when no keypoints are found.  I've only tested with 'GFTT' created by the `imutils` factory because from it looks like thats the only place `corners_to_keypoints` is used.

```python
def corners_to_keypoints(corners):
    """function to take the corners from cv2.GoodFeaturesToTrack and return cv2.KeyPoints"""
    if corners is None:
        keypoints = []
    else:
        keypoints = [cv2.KeyPoint(kp[0][0], kp[0][1], 1) for kp in corners]

    return keypoints
```

Edit: converted output from `None` to `[]` to be consistent with other keypoint detectors.

**Comment by abhiTronix** [comment]
@AdamSpannbauer i always tweak my code for 'NoneType' errors . Basic stuff

**Comment by jrosebr1** [comment]
I agree with @AdamSpannbauer. We can always use `if` or `try` logic to catch when there are no keypoints returned by GFTT but it would be better to have `corners_to_keypoints` return an empty list as that is the behavior of all other keypoint detectors. 

I have merged in #72 to address and resolve the issue. Thank you for the contribution Adam!
